adapters/Line/LineAdapter/GADMediationAdapterLine.m

```diff
@@ -14,23 +14,104 @@
+    GADMediationAdapterLineMutableSetAddObject(
+        applicationIDSet, credentials.settings[GADMediationAdapterLineCredentialKeyApplicationID]);
+  }
+
+  if (!applicationIDSet.count) {
+    *errorPtr = GADMediationAdapterLineErrorWithCodeAndDescription(
+        GADMediationAdapterLineErrorInvalidServerParameters,
+        @"Server configuration did not contain any application ID for LINE mediation.");
+    return nil;
+  }
+
+  NSString *applicationID = applicationIDSet.anyObject;
+  if (applicationIDSet.count > 1) {
+    GADMediationAdapterLineLog(@"Found multiple application IDs. Please remove unused application "
+                               @"IDs from the AdMob UI. Application IDs: %@",
+                               applicationIDSet);
+    GADMediationAdapterLineLog(@"Initializing FiveAd SDK with the application ID: %@",
+                               applicationID);
+  }
+  return applicationID;
+}
+
 @implementation GADMediationAdapterLine
 
 + (GADVersionNumber)adapterVersion {
   GADVersionNumber version = {0};
-  // TODO: configure with appropriate adapter version.
+  NSArray<NSString *> *components =
+      [GADMediationAdapterLineVersion componentsSeparatedByString:@"."];
+  if (components.count == 4) {
+    version.majorVersion = components[0].integerValue;
+    version.minorVersion = components[1].integerValue;
+    version.patchVersion = components[2].integerValue * 100 + components[3].integerValue;
+  }
   return version;
 }
 
 + (GADVersionNumber)adSDKVersion {
+  NSString *versionString = [FADSettings semanticVersion];
+  NSArray<NSString *> *versionComponents = [versionString componentsSeparatedByString:@"."];
+
   GADVersionNumber version = {0};
-  // TODO: configure with appropriate FiveAd SDK's version.
+  if (versionComponents.count >= 3) {
+    version.majorVersion = versionComponents[0].integerValue;
+    version.minorVersion = versionComponents[1].integerValue;
+    version.patchVersion = versionComponents[2].integerValue;
+  }
   return version;
 }
 
 + (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
-  // TODO: return appropriate extra class.
-  return nil;
+  return Nil;
+}
+
++ (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
+             completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
+  NSCAssert(completionHandler, @"Completion handler must not be nil.");
+
+  if (FADSettings.isConfigRegistered) {
+    GADMediationAdapterLineLog(@"FiveAd SDK is already registered");
+    completionHandler(nil);
+    return;
+  }
+
+  NSError *error = nil;
+  NSString *applicationID = GADMediationAdapterLineApplicationID(configuration, &error);
+  if (error) {
+    completionHandler(error);
+    return;
+  }
+
+  // Initialize FiveAd SDK.
+  GADMobileAds *mobileAds = GADMobileAds.sharedInstance;
+  FADConfig *config = [[FADConfig alloc] initWithAppId:applicationID];
+  [config enableSoundByDefault:mobileAds.applicationMuted];
```

Comment:
I believe `enableSoundByDefault` argument should be inverted of `mobileAds.applicationMuted`. 

```
[config enableSoundByDefault:!mobileAds.applicationMuted];
```

Reply:
ah yes, thank you for catching it.